Add section documenting usage and requirements for `terragrunt` (#198)
minamijoyo/tfmigrate: README.md

The minimum required version is OpenTofu v1.6 or higher.

### Terragrunt

#### Without dynamic state

> If you are not leveraging `terragrunt`s [dynamic state generation](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#remote_state) the environment variable `TF_MIGRATE_EXEC_PATH` must be set to `terragrunt`.

Reviewer: Missing apostrophe `'`

```suggestion
If you are not leveraging terragrunt's [dynamic state generation](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#remote_state) the environment variable `TF_MIGRATE_EXEC_PATH` must be set to `terragrunt`.
```

```shell
# As part of the command or via exporting the variable to your shell. 
TFMIGRATE_EXEC_PATH=terragrunt tfmigrate $OTHEROPTIONS
```

#### With dynamic state

> If you are leveraging `terragrunt`s [dynamic state generation](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#remote_state), the `remote_state` block must include a `generate` block.
> 
> This ensures that that `terragrunt` doesn't utilize command line flags for remote state configuration that are incompatible with the local backend, which is utilized by `tfmigrate` for planning.

Reviewer: A suggestion for clarity:

```suggestion
As tfmigrate uses the local backend for planning, some command line flags for the remote state backend cannot be used. If you are leveraging terragrunt's [dynamic state generation](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#remote_state), the `remote_state` block must include a `generate` block.```

> ```

Reviewer: Set file type for the exmaple
```suggestion
```hcl
```
